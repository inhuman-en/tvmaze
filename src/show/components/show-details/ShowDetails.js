import React, {  } from 'react';
import './ShowDetails.scss';
import SanitizedHTML from 'react-sanitized-html';

const ShowDetails = ({ data }) => {
  return <div>
    <div>{data.name}</div>
    <div>
    <SanitizedHTML html={data.summary} />
    </div>
    <a href={data.image.original} rel="noreferrer" target="_blank">
      <img src={data.image.medium} alt={data.name} title={data.name} />
    </a>
    </div>
};

export default ShowDetails;
