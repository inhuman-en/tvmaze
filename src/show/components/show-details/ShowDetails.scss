@import 'style/variables';
@import 'style/mixins';

$nav__height: 30px;
$margin: 15px;
$poster-width: 210px;
$poster-height: 295px;

.show-details {
  display: flex;
  flex-direction: column;
  &__title {
    font-size: $font__size--huge;
    font-weight: $thin;
  }

  &__main {
    display: flex;
    &__poster {
      display: block;
      margin: $margin $margin $margin 0;
      text-align: center;
      min-width: 210px;
      min-height: 295px;
    }

    &__description {
      line-height: 20px;
      > * {
        margin: 0;
        line-height: 25px;
        text-align: justify;
      }
    }
  }

  @include md {
    align-items: center;
    &__main {
      flex-direction: column;
    }
  }
}

