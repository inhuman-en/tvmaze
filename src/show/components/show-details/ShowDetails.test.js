import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ShowDetails from './ShowDetails';

configure({ adapter: new Adapter() });

describe('ShowDetails component', () => {
  const mockProps = {
    data: {
      image: {
        medium: 'https://example.com/medium',
        original: 'https://example.com/original',
      },
      name: 'show name',
      summary: 'show summary',
    },
  };

  it('should render', () => {
    const result = toJson(shallow(<ShowDetails {...mockProps} />));
    expect(result).toMatchSnapshot();
  });
});
