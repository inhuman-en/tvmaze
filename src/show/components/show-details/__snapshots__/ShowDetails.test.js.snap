// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowDetails component should render 1`] = `
<div
  className="show-details"
>
  <div
    className="show-details__title"
  >
    show name
  </div>
  <div
    className="show-details__main"
  >
    <a
      className="show-details__main__poster"
      href="https://example.com/original"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="show name"
        src="https://example.com/medium"
        title="show name"
      />
    </a>
    <div
      className="show-details__main__description"
    >
      <SanitizedHTML
        allowedTags={
          Array [
            "p",
            "b",
          ]
        }
        html="show summary"
      />
    </div>
  </div>
</div>
`;
